core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -720,6 +718,34 @@ class QualificationAppInfo(
       appId, hasFailures)
   }
 
+  def getAllSQLDurations: Seq[Long] = {
+    sqlIdToInfo.flatMap { case (_, info) =>
+      info.rootExecutionID match {
+        // We return the duration if sqlId doesn't have a rootExecutionID or if the rootExecutionID
+        // is the same as the sqlId. In some cases, the child completes the execution before the
+        // parent, so we need to check if the child's duration is within the parent's duration.
+        // We add the child's duration to the parent's duration if it's not within the parent's
+        // duration.
+        case Some(rootExecutionID) if rootExecutionID != info.sqlID =>
+          sqlIdToInfo.get(rootExecutionID).flatMap { rootExecutionInfo =>
+            val rootExecutionStartTime = rootExecutionInfo.startTime
+            val rootExecutionEndTime = rootExecutionInfo.endTime.getOrElse(0L)
+            val sqlStartTime = info.startTime
+            val sqlEndTime = info.endTime.getOrElse(0L)
+            //  Below check will be true if the child is not completely inside the root.
+            //  Nevertheless we still account for its total duration and not the overlap.
```

Comment:
Can this cause extra calculation? I wanted to understand why are we not using only the overlap duration.

Reply:
Wanted to keep the durations consistent based on SQL-ID's. If the user wants to verify the durations from the output file(s), then the results may not match. From the testing, the sqlId's are either completely inside the rootID's or are completely outiside. But there could be some cases where we might include extra durations(of the overlap).